lib/oidcstrategy.js

```diff
@@ -69,18 +68,22 @@ const NONCE_MAX_AMOUNT = 10;
 const NONCE_LIFE_TIME = 3600; // second
 
 function makeProfileObject(src, raw) {
+  var emails = [];
+  if (src.upn)
+    emails = [src.upn];
+  else if (src.emails)
+    emails = src.emails;
```

Comment:
Confirmed with Danny, AAD won't return `email` claim in id_token. To get `email` claim, one needs to call graph api.